src/sagemaker_xgboost_container/algorithm_mode/train.py

```diff
@@ -242,14 +246,27 @@ def train_job(train_cfg, train_dmatrix, val_dmatrix, train_val_dmatrix, model_di
             num_class = train_cfg.get("num_class", None)
             objective = train_cfg.get("objective", None)
             # RepeatedStratifiedKFold expects X as array-like of shape (n_samples, n_features)
-            X = range(train_val_dmatrix.num_row())
-            y = train_val_dmatrix.get_label() if num_class or objective.startswith("binary:") else None
+            classification_problem = num_class or objective.startswith("binary:")
+            num_rows_in_dataset = train_val_dmatrix.num_row()
+            X = range(num_rows_in_dataset)
+            y = train_val_dmatrix.get_label() if classification_problem else None
             rkf = RepeatedStratifiedKFold(n_splits=kfold, n_repeats=num_cv_round) if y is not None \
                 else RepeatedKFold(n_splits=kfold, n_repeats=num_cv_round)
 
-            for train_index, val_index in rkf.split(X=X, y=y):
+            # For regression, ground truth and predicted values are stored. For classification, additionally
+            # estimated probability for predicted class is stored. Predictions are aggregated across
+            # different cross validation repeats - averaged for regression, mode is calculated for classification.
+            data_to_store = np.zeros((num_rows_in_dataset, num_cv_round, 3 if classification_problem else 2))
```

Comment:
Would naming this `postprocessing_data` be accurate? Just to be explicit that this data includes data that is not used for training.

Reply:
Yup that is more accurate